cmd/dump.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-package/internal/cobraext"
+	"github.com/elastic/elastic-package/internal/dump"
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+)
+
+const dumpLongDescription = `Use this command as a exploratory tool to dump assets relevant for the package.`
+
+const dumpInstalledObjectsLongDescription = `Use this command to dump objects installed by Fleet as part of a package.
+
+Use this command as a exploratory tool to dump objects as they are installed by Fleet when installing a package. Dumped objects are stored in files as they are returned by APIs of the stack, without any processing.`
+
+func setupDumpCommand() *cobraext.Command {
+	dumpInstalledObjectsCmd := &cobra.Command{
+		Use:   "installed-objects",
+		Short: "Dump objects installed in the stack",
+		Long:  dumpInstalledObjectsLongDescription,
+		RunE:  dumpInstalledObjectsCmd,
+	}
+	dumpInstalledObjectsCmd.Flags().StringP(cobraext.DumpPackageFlagName, "p", "", cobraext.DumpPackageFlagDescription)
+	dumpInstalledObjectsCmd.MarkFlagRequired(cobraext.DumpPackageFlagName)
+	dumpInstalledObjectsCmd.Flags().StringP(cobraext.DumpOutputFlagName, "o", "output", cobraext.DumpOutputFlagDescription)
+	dumpInstalledObjectsCmd.Flags().Bool(cobraext.TLSSkipVerifyFlagName, false, cobraext.TLSSkipVerifyFlagDescription)
+
+	cmd := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump package assets",
+		Long:  dumpLongDescription,
+	}
+
+	cmd.AddCommand(dumpInstalledObjectsCmd)
+
+	return cobraext.NewCommand(cmd, cobraext.ContextGlobal)
+}
+
+func dumpInstalledObjectsCmd(cmd *cobra.Command, args []string) error {
+	packageName, err := cmd.Flags().GetString(cobraext.DumpPackageFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.DumpPackageFlagName)
+	}
+
+	outputPath, err := cmd.Flags().GetString(cobraext.DumpOutputFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.DumpOutputFlagName)
+	}
+
+	tlsSkipVerify, _ := cmd.Flags().GetBool(cobraext.TLSSkipVerifyFlagName)
+
+	var clientOptions []elasticsearch.ClientOption
+	if tlsSkipVerify {
+		clientOptions = append(clientOptions, elasticsearch.OptionWithSkipTLSVerify())
+	}
+	client, err := elasticsearch.Client(clientOptions...)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize Elasticsearch client")
+	}
+
+	dumper := dump.NewInstalledObjectsDumper(client.API, packageName)
+	n, err := dumper.DumpAll(cmd.Context(), outputPath)
+	if err != nil {
+		return errors.Wrap(err, "dump failed")
+	}
+	if n == 0 {
+		cmd.Printf("No objects were dumped for package %s, is it installed?", packageName)
```

Comment:
nit: do you think we can check if the package is installed apriori? I hope there is a fleet API to make sure it is.

Reply:
Maybe, but this way this tool can also discover letfovers after uninstalling a package :) Also adding a check against the fleet API adds an additional dependency. As the command is now, it only needs access to Elasticsearch.

Reply:
> Maybe, but this way this tool can also discover letfovers after uninstalling a package

It's a discussion whether it's a bug or feature :) We can keep it as is for now.